selfservice/strategy/oidc/provider_apple.go

```diff
@@ -146,3 +147,21 @@ func decodeQuery(query url.Values, claims *Claims) {
 		}
 	}
 }
+
+var _ IDTokenVerifier = new(ProviderApple)
+
+func (a *ProviderApple) Verify(ctx context.Context, rawIDToken string) (*Claims, error) {
+	keySet := oidc.NewRemoteKeySet(ctx, "https://appleid.apple.com/auth/keys")
+	verifier := oidc.NewVerifier("https://appleid.apple.com", keySet, &oidc.Config{
+		ClientID: a.config.ClientID,
+	})
+	token, err := verifier.Verify(ctx, rawIDToken)
+	if err != nil {
+		return nil, err
+	}
+	claims := &Claims{}
+	if err := token.Claims(claims); err != nil {
+		return nil, err
+	}
+	return claims, nil
```

Comment:
Any ideas on how to test this are appreciated. I thought about adding a test that just uses a real `idToken` and verifies that, but that token contains my personal email address and user ID, which I'd rather not leak here. :)

I guess we could create a dummy Apple ID, obtain the token and add that here, however we would need to refresh the token, as Apple rotates their JWKs. Also, not sure how to deal with expiry.

Reply:
Generate a JWT with the same claims as apple, and set up a fake HTTP server with a JWK you control - then verify. That should work!

Reply:
Yup, that works. I needed to make the JWKS URL configurable via a private field, which defaults to the apple URL.